src/qililab/instruments/qblox/qblox_qrm.py

```diff
@@ -187,6 +187,33 @@ def acquire_result(self) -> QbloxResult:
         """
         return self.get_acquisitions()
 
+    def acquire_qprogram_results(self, acquisitions: list[str]) -> list[QbloxQProgramMeasurementResult]:  # type: ignore
```

Comment:
Another worry I have: beware that for the old workflow we delete the acquisition data of old programs. We need to do this, otherwise the old data is still stored in each bin, and will be averaged with the new data!

Reply:
With Victor we started thinking about what are all the steps (for Qblox) that we need to do everytime we execute a new program/circuit. This would be a good time to write them down!

Reply:
> Another worry I have: beware that for the old workflow we delete the acquisition data of old programs. We need to do this, otherwise the old data is still stored in each bin, and will be averaged with the new data!

Can you point me to the line that executes this in the old workflow?

Reply:
Here: https://github.com/qilimanjaro-tech/qililab/blob/83b6ea22f91b10da945a61f99ba56132c585c709/src/qililab/instruments/qblox/qblox_qrm.py#L178

Reply:
Wait, maybe by re-uploading a program the acquisition data is removed...

Reply:
Yes, you are right! Missed that! I added this to the method and tests.

Reply:
Did you test it in HW? Because now I think we don't need it 😆 

If you check the line above, the deletion is done only when we don't re-upload the sequence.

Given that in the QProgram case we always upload a program, maybe we don't even need this...

Reply:
I cannot test in HW again until Monday!

The line above though, is run if a sequence was previously uploaded, not if we re-upload one. So it runs every time after uploading a sequence.

Reply:
I believe your original thought on needing this is correct.

Reply:
I'm not so sure... 🤔 We can discuss this after the daily if you want!

Reply:
also it's probably safer to do
`self.device.delete_acquisition_data(sequencer=sequencer.identifier, all=True)` which clears all acquisition data in the sequencer. Note that deleting acquisitions only sets all the acquisition values to `None` but the sequencer still has the acquisition data entries (which in this case will be something like `{"default: [None, None, ...]}`)